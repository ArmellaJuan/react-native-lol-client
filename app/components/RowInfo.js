import React, { Component, PropTypes } from 'react';
import { StyleSheet, View, Text } from 'react-native';


export default class RowInfo extends Component{

  constructor(props){
    super(props);
  }

  render() {
    return (
        <View style = {styles.rowInfo} >
          <Text style = { [styles.rowLabel,  this.props.fontSize? { fontSize: this.props.fontSize } : {}, this.props.labelStyle? this.props.labelStyle : {}  ] }>{this.props.label}</Text>
          <Text style = { [styles.rowValue, this.props.fontSize? { fontSize: this.props.fontSize } : {}, this.props.valueStyle? this.props.valueStyle : {} ] }>{this.props.value}</Text>
        </View>
    );
  }

}


RowInfo.propTypes = {
  label: PropTypes.string,
  value: PropTypes.node,
  fontSize: PropTypes.number,
  valueStyle: PropTypes.array,
  labelStyle: PropTypes.array
};



const styles = StyleSheet.create({
  rowInfo: {
    flexDirection: 'row',
    height: 35,
    justifyContent: 'center',
    alignItems: 'center'
  },
  rowLabel: {
    flex:1,
    fontWeight: 'bold',
  },
  rowValue: {
    flex:1,
  },
});