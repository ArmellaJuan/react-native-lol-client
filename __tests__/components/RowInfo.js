import 'react-native';
import React from 'react';
import RowInfo from '../../app/components/RowInfo';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

test('renders correctly', () => {

  const tree = renderer.create(
    <RowInfo value="TestValue" label="TestLabel" />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});