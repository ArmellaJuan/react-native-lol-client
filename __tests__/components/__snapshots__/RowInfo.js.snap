exports[`test renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "height": 35,
      "justifyContent": "center",
    }
  }>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "flex": 1,
          "fontWeight": "bold",
        },
        Object {},
        Object {},
      ]
    }>
    TestLabel
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "flex": 1,
        },
        Object {},
        Object {},
      ]
    }>
    TestValue
  </Text>
</View>
`;
